enhancements/sig-architecture/30-clusterset-override/README.md

```diff
@@ -0,0 +1,239 @@
+- I want to create a managedClusterSet for Disaster Recovery purposes, the managedClusterSet should select two or more clusters in different regions.
+- I want to create a managedClusterSet to group clusters based on the middleware enabled on these clusters, so I could run special applications in these clusters.
+ 
+## Proposal
+1. Changes on ManagedClusterSet APIs.
+ 
+Add a field into ManagedClusterSet spec
+ 
+```go
+type ManagedClusterSetSpec struct {
+ // ExclusiveKey defines the key of the label which the managedClusterSet will use to select managedCluster.
+ // The managedClusterSet will select ManagedClusters based on the managedCluster's label "<ExclusiveKey>:<ManagedClusterSetName>".
+ // If the exclusiveKey is not set, we will use `cluster.open-cluster-management.io/clusterset` as exclusiveKey.
+ // The value can not change once the managedClusterSet is created.
+ // +optional
+ ExclusiveKey string `json:"exclusiveKey"`
+}
+```
+ 
+Notes:
+As managedClusterSet uses the label `<ExclusiveKey>:<ManagedClusterSetName>` to filter managedclusters. So if I want to use one label key as exclusiveKey, the label value must match the k8s name rule.
+ 
+2. RBAC change
+ 
+Given the flexibility introduced by exclusive key, instead of control whether a cluster can be added into a managedClusterSet with sub resource of `managedclustersets/join`, it might make more sense to control whether a user can create a managedClusterSet with a certain exclusive key. We can define a subresource of `managedclustersets/exclusiveKey`, to control the creation of certain managedClusterSet.
+ 
+I want to create the managedClusterSet `apac`, I must have the following permission.
+ 
+```yaml
+apiVersion: cluster.open-cluster-management.io/v1alpha1
+kind: ManagedClusterSet
+metadata:
+name: apac
+spec:
+exclusiveKey: region
+```
+ 
+```yaml
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+ name: clustersetrole
+rules:
+ - apiGroups: ["cluster.open-cluster-management.io"]
+   resources: ["managedclustersets/exclusiveKey"]
+   resourceNames: ["region"]
+   verbs: ["create"]
+ - apiGroups: ["cluster.open-cluster-management.io"]
+   resources: ["managedclustersets"]
+   resourceNames: ["apac"]
+   verbs: ["*"]
+```
+ 
+ 
+## Examples
+### I want to create a managedClusterSet which include all clusters in apac region. Then I can apply certain configurations to all clusters in apac.
+1. For each cluster, I need to set a label `region=apac` for all clusters in the apac region.
+2. I need to get the following permission to create a managedClusterSet.
+```yaml
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+ name: clustersetrole
+rules:
+ - apiGroups: ["cluster.open-cluster-management.io"]
+   resources: ["managedclustersets/exclusiveKey"]
+   resourceNames: ["region"]
+   verbs: ["create"]
+ - apiGroups: ["cluster.open-cluster-management.io"]
+   resources: ["managedclustersets"]
+   resourceNames: ["apac"]
+   verbs: ["*"]
+```
```

Comment:
I don't think you need to repeat the required RBAC rules for each example because they have already been covered in the above section.